packages/ui/src/molecules/QuantitySelector/QuantitySelector.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { render } from '@testing-library/react'
+import React, { useState } from 'react'
+import userEvent from '@testing-library/user-event'
+import { axe } from 'jest-axe'
+
+import QuantitySelector from '.'
+
+const QuantitySelectorTest = () => {
+  const [quantity, setQuantity] = useState(1)
+
+  return (
+    <QuantitySelector
+      name="quantity-selector"
+      quantity={quantity}
+      leftButtonProps={{
+        icon: <span>-</span>,
+        onClick: () => {
+          setQuantity((curr) => curr - 1)
+        },
+      }}
+      rightButtonProps={{
+        icon: <span>+</span>,
+        onClick: () => {
+          setQuantity((curr) => curr + 1)
+        },
+      }}
+      inputProps={{ readOnly: true }}
+    />
+  )
+}
+
+describe('QuantitySelector', () => {
+  it('should render QuantitySelector with its children', () => {
+    const { getByTestId } = render(
+      <QuantitySelector
+        name="quantity-selector"
+        quantity={1}
+        leftButtonProps={{ icon: <span>-</span> }}
+        rightButtonProps={{ icon: <span>+</span> }}
+        inputProps={{ readOnly: true }}
+      />
+    )
+
+    expect(getByTestId('store-quantity-selector')).toBeInTheDocument()
+  })
+
+  it('should render QuantitySelector with the correct quantity', () => {
+    const { getByLabelText } = render(
+      <QuantitySelector
+        name="quantity-selector"
+        quantity={123}
+        leftButtonProps={{ icon: <span>-</span> }}
+        rightButtonProps={{ icon: <span>+</span> }}
+        inputProps={{ readOnly: true }}
+      />
+    )
+
+    expect(getByLabelText('Quantity')).toHaveValue('123')
+  })
+
+  it('Should update correctly according to event changes', () => {
+    const { getByLabelText } = render(<QuantitySelectorTest />)
```

Comment:
Not sure I like how _far away_ the `QuantitySelectorTest` component definition is from its only use (line 8). I'd prefer to inline it inside this test so it's self-contained and not spread throughout the file. Just some thought!

Reply:
I've moved the definition to the test scope and made some other adjustments at https://github.com/vtex/faststore/pull/1083/commits/8bf6b2ad91023a6d7922570106bee2d06356ddab